packages/react-meteor-data/useTracker.ts

```diff
@@ -103,71 +108,94 @@ const useTrackerNoDeps = <T = any>(reactiveFn: IReactiveFn<T>) => {
   return refs.trackerData;
 }
 
-const useTrackerWithDeps = <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList): T => {
-  let [data, setData] = useState<T>();
-
-  const { current: refs } = useRef({ reactiveFn });
+const useTrackerWithDeps = <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList, skipUpdate: ISkipUpdate<T> = null): T => {
+  const [data, setData] = useState<T>();
+  const { current: refs } = useRef({ reactiveFn, data });
   refs.reactiveFn = reactiveFn;
+  refs.data = data;
```

Comment:
it doesn't work, because after rerender data === undefined, which leads to refs.data === undefined.

Reply:
I probably need to change this line:
https://github.com/meteor/react-packages/blob/e30e2905f8bcfca095f5929ef9888b061b39584a/packages/react-meteor-data/useTracker.ts#L115

to something like:

```js
if (typeof refs.data !== "undefined") {
  refs.data = data;
}
```

Reply:
We should also write a unit test for this, seems it would be easy enough

Reply:
I wrote a test to make sure immediate rerenders (calling a state setter or similar, during render) wouldn't reset the value. Without making this change, the test passes. I could use more help understanding how to trigger the problem you see:

```js

  Tinytest.addAsync('useTracker - immediate rerender does not result in `undefined`', async function (test, completed) {
    const container = document.createElement("DIV");
    const reactiveDict = new ReactiveDict();
    let value;
    let renders = 0;
    const Test = () => {
      renders++;
      const [num, setNum] = useState(0);
      value = useTracker(() => {
        reactiveDict.setDefault('key', 'initial');
        return reactiveDict.get('key');
      });
      if (num === 0) {
        setNum(1);
      }
      return <span>{value}</span>;
    };

    ReactDOM.render(<Test />, container);
    test.equal(value, 'initial', 'values should be "initial" and not undefined');
    test.equal(renders, 2, "Should have rendered twice");

    // wait for useEffect
    await waitFor(() => {}, { container, timeout: 250 });

    test.equal(value, 'initial', 'values should still be "initial" after mount');
    test.equal(renders, 3, "Should have rendered thrice");

    completed();
  });
```

Reply:
Oh! I forgot to add the deps array, heh. Once I did that, I did indeed, confirm the bug!

The test and the hook are now fixed. Thanks!